### ISSUE TITLE
TaskMon crashes when trying to sort by CPU

### ISSUE THREAD

**Issue by Menelion** [trypsynth/taskmon#8]
At least on my system it's 100% reproducible. Windows 11 X64, JAWS 2026.

1. Launch TaskMon with no taskmon.ini present.
2. Press Shift+Tab.
3. Press right Arrow twice to land on CPU.
4. Boom!

And the next time I start the program, it's sorted by PID, not by CPU.

**Comment by trypsynth** [comment]
Unable to reproduce with both NVDA and Narrator on Windows 11 24H2, I don't have JAWS. Is this behavior exclusive to JAWS?

**Comment by jcsteh** [comment]
I... was able to reproduce this. The crash is a stack overflow. However, once I got a build in place with debug symbols so I could debug it, it stopped happening. I'll keep that build in place in case this shows up again. 😕

**Comment by trypsynth** [comment]
I didn't actually want to close this as I can't confirm it's fixed, can someone here please try the latest commit?

**Comment by jcsteh** [comment]
Oh, heh, I didn't see this before I debugged. I'll try the latest commit in a sec. For reference, here's the stack dump:

```
(4354.2ee4): Stack overflow - code c00000fd (first chance)
First chance exceptions are reported before any exception handling.
This exception may be expected and handled.
*** WARNING: Unable to verify checksum for taskmon.exe
taskmon!quicksort+0x45:
00007ff7`5bd33e01 4889442430      mov     qword ptr [rsp+30h],rax ss:00000079`db243310=0000000000000000
0:000> kpn
 # Child-SP          RetAddr               Call Site
00 00000079`db2432e0 00007ff7`5bd33efb     taskmon!quicksort(struct process_entry * entries = 0x0000018f`9b976fd0, int low = 0n0, int high = 0n21, sort_field field = SORT_FIELD_CPU (0n2), int descending = 0n0)+0x45 [C:\Users\jamie\src\taskmon\src\process.c @ 272] 
01 00000079`db2442d0 00007ff7`5bd33efb     taskmon!quicksort(struct process_entry * entries = 0x0000018f`9b976fd0, int low = 0n0, int high = 0n22, sort_field field = SORT_FIELD_CPU (0n2), int descending = 0n0)+0x13f [C:\Users\jamie\src\taskmon\src\process.c @ 283] 
...
f9 00000079`db33b350 00007ff7`5bd33efb     taskmon!quicksort(struct process_entry * entries = 0x0000018f`9b976fd0, int low = 0n0, int high = 0n270, sort_field field = SORT_FIELD_CPU (0n2), int descending = 0n0)+0x13f [C:\Users\jamie\src\taskmon\src\process.c @ 283] 
fa 00000079`db33c340 00007ff7`5bd33efb     taskmon!quicksort(struct process_entry * entries = 0x0000018f`9b976fd0, int low = 0n0, int high = 0n271, sort_field field = SORT_FIELD_CPU (0n2), int descending = 0n0)+0x13f [C:\Users\jamie\src\taskmon\src\process.c @ 283] 
fb 00000079`db33d330 00007ff7`5bd33efb     taskmon!quicksort(struct process_entry * entries = 0x0000018f`9b976fd0, int low = 0n0, int high = 0n272, sort_field field = SORT_FIELD_CPU (0n2), int descending = 0n0)+0x13f [C:\Users\jamie\src\taskmon\src\process.c @ 283] 
fc 00000079`db33e320 00007ff7`5bd34da8     taskmon!quicksort(struct process_entry * entries = 0x0000018f`9b976fd0, int low = 0n0, int high = 0n286, sort_field field = SORT_FIELD_CPU (0n2), int descending = 0n0)+0x13f [C:\Users\jamie\src\taskmon\src\process.c @ 283] 
fd 00000079`db33f310 00007ff7`5bd329f4     taskmon!snapshot_processes(struct snapshot_entry * snapshots = 0x00007ff7`5bd3a2c0, int * out_count = 0x00000079`db33f930, sort_field field = SORT_FIELD_CPU (0n2), int descending = 0n0)+0xd70 [C:\Users\jamie\src\taskmon\src\process.c @ 576] 
fe 00000079`db33f780 00007ff7`5bd362ab     taskmon!do_refresh(void)+0x50 [C:\Users\jamie\src\taskmon\src\listview.c @ 241] 
ff 00000079`db33f930 00007fff`5f4b66b5     taskmon!sort_btn_proc(struct HWND__ * hwnd = <Value unavailable error>, unsigned int msg = <Value unavailable error>, unsigned int64 wp = <Value unavailable error>, int64 lp = <Value unavailable error>, unsigned int64 id = 1, unsigned int64 data = 0)+0x1db [C:\Users\jamie\src\taskmon\src\sortbar.c @ 56] 
```

I think this only occurs when higher optimisation settings are enabled because the function gets inlined in that case, so the stack blows out more when it recurses. With RelWithDebInfo, this didn't occur, probably because it optimises less. For reference, here's the patch I used to get a Release build to provide useful symbols without changing the characteristics of the build:

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index a61d7ea..7091b0d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,8 +5,8 @@ file(GLOB SOURCES CONFIGURE_DEPENDS "src/*.c" "src/*.rc")
 add_executable(taskmon WIN32 ${SOURCES})
 target_compile_definitions(taskmon PRIVATE UNICODE _UNICODE WIN32_LEAN_AND_MEAN NOMINMAX)
 if(MSVC)
-	target_compile_options(taskmon PRIVATE /GS- /W4 /Os /Oy)
-	target_link_options(taskmon PRIVATE /NODEFAULTLIB /ENTRY:WinMainCRTStartup /SUBSYSTEM:WINDOWS /OPT:REF /OPT:ICF /MERGE:.rdata=.text /MERGE:.pdata=.text)
+	target_compile_options(taskmon PRIVATE /GS- /W4 /Os /Oy /Zi)
+	target_link_options(taskmon PRIVATE /NODEFAULTLIB /ENTRY:WinMainCRTStartup /SUBSYSTEM:WINDOWS /OPT:REF /OPT:ICF /MERGE:.rdata=.text /MERGE:.pdata=.text /debug:full )
 endif()
 target_link_libraries(taskmon PRIVATE kernel32 user32 gdi32 shell32 comctl32 ole32 shlwapi ntdll advapi32 dwmapi uxtheme comdlg32 version)
 
```

**Comment by jcsteh** [comment]
Yup, 0185e0d fixes the issue for me. Thanks.

**Comment by Menelion** [comment]
@trypsynth sorry for the delay. confirmed fixed, thanks!
